### ISSUE TITLE
BungeeCord Integration

### ISSUE THREAD

**Issue by HeroCC** [Scarsz/DiscordSRV#34]
Hello!

I currently run a BungeeCord server, and would love to integrate with discord. However, I don't want to put DiscordSRV in every server, and would rather just to put it in bungee. I use VentureChat, so it's integration drew me to this plugin vs others that support bungee but not VC.

If you have no plans for BungeeCord that is fine, but it'd be nice to have one less plugin to put in every server / sync configs. 

Thanks!

**Comment by Scarsz** [comment]
I would but honestly a grand total of two people want this, there's just no
motivation to make it currently

**Comment by HeroCC** [comment]
That's OK. If you ever change your mind, know that it would be appreciated!

**Comment by Scarsz** [comment]
it's not a no, it's just a rather low priority right now for me

**Comment by InfinityAspire** [comment]
Even though it's in the development queue I'd like to say I'm holding out for this feature too

**Comment by Eufranio** [comment]
Same.

**Comment by RedEyedRocker** [comment]
I'm running Bungeecord server as well so I request for this feature as well.

**Comment by AaronMcHale** [comment]
+1

Also HeroChat Integration could continue in a Bungee version with https://www.spigotmc.org/resources/herochatbungeebridge-bungee.189/

**Comment by Carisus** [comment]
I would like to see Bungee support as well. The Ability to see which server they are talking from without setting up multiple Discord channels would be a major plus.

**Comment by haugsvar** [comment]
I second a Bungee version

**Comment by Goofables** [comment]
I agree. This would be a nice feature

**Comment by jonthesquirrel** [comment]
This would be invaluable for me. Especially since one of my servers behind bungee is vanilla, and a bungeecord-only discord plugin is the only option.

**Comment by PenguinNexus** [comment]
Same. I found that doing /discord link works in my creative server but doesn't carry over to my survival server as it's two seperate plugins. Please put this in!

**Comment by SimplyBallistic** [comment]
I was looking into developing a completely new plugin called Disgot that provided Bungeecord support, but instead of making a whole new plugin, should I instead make a fork addressing this issue?

**Comment by Scarsz** [comment]
@SimplyBallistic If you want to contribute to DiscordSRV instead it would be much appreciated, PM me on Discord https://scarsz.me/discord

**Comment by SuperRonanCraft** [comment]
I would love to support this idea, the only bungeecord to discord only involves login and logout times, not the actual console log or commands

**Comment by Bezod** [comment]
I love would love this feature as well!

**Comment by SimplyBallistic** [comment]
Im having trouble understanding what is needed. What exactly would a bungee cord version do? What should it do? What should it do compared to the current version. The current version seems fine to me.

**Comment by SuperRonanCraft** [comment]
@SimplyBallistic it would do the same thing as the Bukkit/Spigot version would do, just on Bungeecord, it would send the console, but just remove the chat channels, basically a console only version? I just like the idea of having all consoles connected on discord without having to go to your hosts platform to restart,stop or execute command. I don't know about you, but it would open doors to quick and easy developing for other developers who work on networks but whose server owners don't want then to have file access or change settings, just execute commands on the console on Discord.

**Comment by HeroCC** [comment]
I feel like there are other ways to have a portable console, without using discord. A discord plugin without chat would seem pretty pointless.

My idea was it would be pretty much the same (integrating with chat plugins, console, and leave joins) but it would only be installed on BungeeCord, instead of each single spigot. Maybe allow routing different bungee connected servers to different channels in discord?

**Comment by KiwiSAMA** [comment]
It will be a nice feature if it support bungeecord

**Comment by SimplyBallistic** [comment]
@HeroCC @SuperRonanCraft Ok, I understand what you mean. I'll implement it into my fork!

**Comment by xDeeKay** [comment]
+1 for BungeeCord support.

**Comment by LeCodeCo** [comment]
+1 also maybe integrate with DeluxeChat Bungee.

**Comment by RedEyedRocker** [comment]
Been 1 year since the Bungeecord support request and it has still not been developed?

**Comment by Scarsz** [comment]
For every comment this issue gets, this will be delayed by a month

**Comment by RedEyedRocker** [comment]
Meh, It's been 4 months that I'm taking Java classes so when I'm skilled enough, I'll do it myself for my own personal use.

**Comment by RedEyedRocker** [comment]
At first you said "I would but honestly a grand total of two people want this, there's just no
motivation to make it currently" and now when more people want this then you're making it low priority.
I don't have any hope for this anymore.

**Comment by xDeeKay** [comment]
I too figured the more people that asked for this the greater chance there would be. Guess pizza can't buy everything.

**Comment by RedEyedRocker** [comment]
@xDeeKay Well, that's what was said by the developer in the past but I guess, he lied.

**Comment by BillyGalbreath** [comment]
I guess you guys are unaware Scarz is in the process of a big move. His new place isn't even fully furnished yet. Give it some time. This is in the works, along with forge support as well. You should check the other branches in this repo before slinging shit around the place. ;)

**Comment by RedEyedRocker** [comment]
@BillyGalbreath Yeah buy why did he say that he'll delay this more if someone comments on here? He could've just said to give him time as he's busy with life.

**Comment by BillyGalbreath** [comment]
Because it's annoying as hell and stressful. This isn't a job for him, it's a side project/hobby he does in his down time. Imagine if the tables were turned. ;)

**Comment by xDeeKay** [comment]
It's annoying as hell to receive support for a feature well sought after? News to me.

**Comment by BillyGalbreath** [comment]
Oh, and not to mention the bot works as-is on bungee environments. The requested support in this ticket is simply moving the workload from the server owners to scarz. He has no obligation or incentive to make this happen, especially when bungee server owners are either a business for profit, or small network for close friends.

**Comment by RedEyedRocker** [comment]
@BillyGalbreath Then he can put up the Bungee version as a premium plugin. That way, he can earn money as well.

**Comment by BillyGalbreath** [comment]
Then it comes a business and comes with obligations. I don't think that is his goal, which is smart, imho. Not to mention dealing with pirates and leakers, etc.

Just calm your tits and enjoy this free software as it is available to you. Orndont. Up to you.

Making back handed threats is not only annoying, but counterproductive for you. ;)

**Comment by RedEyedRocker** [comment]
@BillyGalbreath It is not a threat. Many people requested the feature. If he doesn't want to develop this then he can just close this issue. He doesn't need to give us excuses like "For every comment this issue gets, this will be delayed by a month"

**Comment by BillyGalbreath** [comment]
I don't believe you have fully read what I've said. I stated he *is* making this feature, and has even already started on it. So stop bugging. There is no ETA. There is no progress report, other than viewing the github branch yourself to see the progress. His reply is not an excuse, it is a vent from frustration out of being annoyed by this ticket, especially since it's a stupid feature request that doesn't need to be made (since the bot work fine on bungee).

I'll use short sentences to recap for you in case that didn't sink in: It's in progress. There is no ETA. So stop pestering.

**Comment by RedEyedRocker** [comment]
@BillyGalbreath No one said it doesn't work on Bungee, the feature is just to remove the unnecessary load on servers, unnecessary copy paste so it will be user friendly. No one is asking for the ETA, people were only supporting this issue like the developer said, he's not motivated to do this as there are not many users who want it. Telling us that this feature would only get delayed a whole month just if someone comments on this issue is just disappointing and doesn't make sense.
This whole thing could've been handled by just saying that he's working on this feature already.
Anyway, I don't have a problem anymore and I don't really care about this feature as I've retired from server development, If I need it, I'll develop it myself for personal use.

**Comment by BillyGalbreath** [comment]
Like I said earlier. This whole ticket is to push the load away from the server owners and into the plugin developer. That's just stupid, especially seeing bungee requires an while rewrite because of the way chat is handled, not even considering all the chat plugin hooks. Anyone ever thought about that? How is a bungee chat pkugin supposed to know which chat messages to intercept if the chat pkugin is on the server instead of proxy? It would require a DiscordSRV bukkit pluginti be installed n each server anyways, and add more load and network ci gestion to communicate icate that data back to the mother instance running on bungee before it can even send the chat to discord... All because you felt lazy and didn't want to copy paste. Good job.

**Comment by RedEyedRocker** [comment]
@BillyGalbreath Then answer me, wouldn't it be more better if it was user friendly? Users having to copy paste the same thing in every single server is unnecessary and no, the point of supporting Bungee means that the config would be on bungeecord and you can just manage the whole plugin from there and the configuration. Putting the DSRV bukkit plugin is just for the chat so it's simple as it wouldn't require you to configure it on every server as it would be done in the bungeecord.
It's not lazy, it's a bad way of managing a plugin just like say, you have duplicate code in a project so you just make a separate method and call that method. Also, the placeholders, question yourself on why they exist and why we use them, for ease of use as if you have tons of lines where there would be the same thing then you can just use placeholders and edit one thing to update everything else which uses it.
The feature is just to make it ease of use and to remove duplicate making it user friendly.

**Comment by BillyGalbreath** [comment]
A.k.a. you're lazy and want scarz to do all the work while you sit back and let your server make a profit. Got it.

**Comment by RedEyedRocker** [comment]
@BillyGalbreath Yeah right, it's difficult to develop an actual server where there are active players and also takes a lot of money as you need to pay for the dedicated 24/7 hosting and stuff. It's even more difficult to get active players and then hope to get donations which won't even be close enough to half the amount of what it costs for a dedicated server. I already have experienced it as I used to work on the server, AcidCraft Network. They made 0$ while have few active members and now, it has been shutdown.
He's a developer, he should enjoy programming. If it's "a lot of work" for him then he can simply decide to not do it in the first place so why you sucking his dick so much? No one here is begging for this feature, they only recommended it because it makes things easier to manage.
I don't have any problem with him developing it or not.

**Comment by Scarsz** [comment]
Locking this thread because of lack of decent discussion. It's being worked on. The entire plugin has to be rewritten and frameworks have to be made for the transition to go smoothly. No ETA. What @BillyGalbreath said is more/less right. WICKED is good.

**Comment by Dinty1** [comment]
Some very bare bones support for BungeeCord and Velocity has been added to testing builds for v2 thanks to @IxPrumxI.

This only covers features such as join, leave, server switch messages, the console and various syncing features.

**Features involving cross-server communication and proxy-server connection are not implemented yet.** 
This also does not allow for using multiple instances of DiscordSRV under one Discord bot (yet) due to conflicts with slash commands.

See https://github.com/DiscordSRV/Ascension#ascension for more info.
